client/trino-client/src/main/java/io/trino/client/auth/external/MemoryCachedKnownToken.java

```diff
@@ -36,13 +37,18 @@
 class MemoryCachedKnownToken
         implements KnownToken
 {
-    public static final MemoryCachedKnownToken INSTANCE = new MemoryCachedKnownToken();
+    private static final ConcurrentHashMap<String, MemoryCachedKnownToken> CACHED_BY_HOST = new ConcurrentHashMap<>();
```

Comment:
This is not a "cache", since there is no invalidation.

Should there be?
Looks like potential of a (rather slow) memory leak.

Reply:
yep it's a memento rather than a cache. We potentially want to keep the tokens as long as they are not rejected by the server, but there is a chance of having an old running driver with abandoned Tokens to a no longer used hosts. This would result in a very slow memory leak. To prevent that, we could use loading cache with some policy of dropping the token after some time, but then the question arises - for how long should we be keeping potentially dropped token. Please mind that we can't really make it configurable, as the purpose of this mechanism is to reuse tokens between all Connections, so it's not playing along with current configuration.

Reply:
> for how long should we be keeping potentially dropped tokens. 

1h?

Reply:
ok I can make it into cache with removeAfterRead(1h), but is it ok to use vanilla Guava Cache? All the goodies with EvictableCache etc. are in plugin toolkit which is not a reasonable dependency for the client.

Reply:
Guava cache is fine as long you don't call `invalidate*` methods. Please suppress the warnings and add proper comment. Eventually you can copy a class to disallow calling `invalidate*` methods.

Reply:
https://github.com/trinodb/trino/pull/10931

Reply:
Thanks, I've changed it to SafeCache and added 1h afterRead drop. Please take a look now @kokosing @findepi